lib/ts/nextjs/types.ts

```diff
@@ -0,0 +1,86 @@
+import type { LoadedSessionContext } from "../recipe/session/types";
+import type { AppInfoUserInput } from "../types";
+
+export type CookiesStore = {
+    get: (name: string) => { value: string };
+    set: (name: string, value: string) => void;
+};
+
+export type CookiesObject = Record<string, string>;
+
+export type GetServerSidePropsRedirect = {
+    redirect: { destination: string; permanent: boolean };
+};
+
+export type GetServerSidePropsReturnValue =
+    | {
+          props: { session: LoadedSessionContext };
+      }
+    | GetServerSidePropsRedirect;
+
+export function isCookiesStore(obj: unknown): obj is CookiesStore {
+    return typeof obj === "object" && obj !== null && "get" in obj && typeof (obj as CookiesStore).get === "function";
+}
+
+export type SuperTokensNextjsConfig = {
+    appInfo: AppInfoUserInput;
+    enableDebugLogs?: boolean;
+};
+
+export type FrontTokenPayload = {
+    uid: string;
+    ate: number;
+    up: {
+        iat: number;
+        exp: number;
+        sub: string;
+        tId: string;
+        rsub: string;
+        sessionHandle: string;
+        refrehTokenHash1: string;
+        parentRefereshTokenHash1: string | null;
+        antiCsrfToken: string | null;
+        iss: string;
+        "st-role": {
+            v: string;
+            t: number[];
+        };
+        "st-perm": {
+            v: string[];
+            t: number;
+        };
```

Comment:
there can be other (even custom) claims in there (for example `st-ev`), do we need to account for that?